### PR TITLE
Implement sunset/sunrise appreciation

### PR DESCRIPTION
**Pull request by BalthazarArgall**
#### Summary
Features "sunset/sunrise appreciation"

#### Purpose of change
Sunsets and sunrises are beautiful yet there's no way to appreciate them in game.

#### Describe the solution
Implement an EoC that triggers at sunrise and sunset and gives a small mood buff.

#### Describe alternatives you've considered

- I would really like to have some snippets to put in the log when the EoC triggers but I'm not great at writing. Maybe someone can implement them later, or give them to me to implement.

#### Testing
Spawned, waited for sunset, got the buff.

#### Additional context
~~Please don't merge this before #74893 or it won't work.~~ Good to go.

### CODE REVIEW

**Comment by Aimless-dude** [comment]
Just spitballing here, but should it only happen in clear, or slightly cloudy weather? There's not much to appreciate when thick stormclouds are blocking your view, after all.

**Comment by BalthazarArgall** [comment]
> Just spitballing here, but should it only happen in clear, or slightly cloudy weather? There's not much to appreciate when thick stormclouds are blocking your view, after all.

I think so, yes. I welcome examples of storms preventing people from seeing the sun rise if you can find some.

**Comment by andrei8l** [comment]
This can cause the effect to trigger at the wrong time if you skip turns past the next occurrence. Perhaps a better way is to recur every few hours and trigger if time_util/since sunset/sunrise is less than 1 hour and some time has passed since last trigger.

**Comment by zachary-kaelan** [comment]
Some snippets totally written by me, a human, and not ChatGPT.

1. **Sunrise Over the Ruins**
   "As the first light crests over the shattered horizon, you can't help but feel a flicker of hope. The dawn's golden embrace warms your face, reminding you that even in the bleakest times, new beginnings are always possible. Your spirits lift, and for a fleeting moment, the weight of survival feels a bit lighter."

2. **Sunset's Respite**
   "The sun dips low, casting long, serene shadows on the remnants of a world once known. The fiery colors of the sky paint a masterpiece of tranquility, as if the universe itself offers a moment of peace. You take a deep breath, feeling the day's trials ease away with the fading light. There's beauty here, even in the aftermath."

3. **Dawn of Resolve**
   "Morning breaks, spreading hues of pink and orange across the desolate landscape. As you watch the sunrise, you find a renewed sense of determination. Each ray of light seems to whisper promises of resilience and strength. Bolstered by the dawn, you know you can face whatever lies ahead."

4. **Evening Solace**
   "The sun sets in a blaze of red and violet, its descent a graceful farewell to the day's struggles. In this quiet moment, the world feels a little less broken. There's comfort in the routine of dusk, a reminder that some things endure. Your heart finds a measure of peace, and your mind unclenches, ready to rest and rebuild."

5. **A New Day's Hope**
   "The sun rises, a brilliant beacon piercing through the remnants of chaos. Each ray of light illuminates the world with a sense of rebirth and endless possibilities. This morning, as you stand amidst the ruins, you feel an irresistible urge to continue, a spark of optimism igniting within. The dawn brings with it a small but vital boost to your morale."

6. **Twilight's Embrace**
   "As the sun sinks beneath the horizon, the sky erupts in a final display of glory. The shadows stretch long, enveloping the ruins in a tranquil embrace. You feel a touch of serenity wash over you, a reminder that endings can be beautiful too. With the night comes rest, and with rest, the strength to continue."

7. **Morning's Promise**
   "The first light of day spills over the landscape, chasing away the shadows of night. Each morning holds the promise of discovery, the chance to start anew. As you watch the sunrise, warmth spreads through your chest—a reassurance that life, even now, finds a way to persist."

8. **End of Day’s Comfort**
   "The sunset bathes the world in hues of amber and crimson, a gentle end to the day's relentless march. In the quiet surrender of nightfall, you find solace. The beauty of the dusk reminds you that not all is lost; there is still grace to be found, even here. This twilight hour grants you a fleeting, precious moment of peace."

**Comment by BalthazarArgall** [comment]
@zachary-kaelan 

I appreciate the help. I don't think I will use AI input as-is but this is giving me some inspiration, I'll try my hand at writing a few snippets myself.

**Comment by zachary-kaelan** [comment]
> I appreciate the help. I don't think I will use AI input as-is but this is giving me some inspiration, I'll try my hand at writing a few snippets myself.

Even if it wasn't AI,those are all overly fancy and wordy. But it can be difficult to find inspiration specifically for writing the thoughts someone has while watching a sunrise or sunset in a post-apocalyptic world.

File: data/json/effects_on_condition/misc_effect_on_condition.json [hunk 1]
```diff
@@ -220,5 +220,39 @@
...
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "sunrise_morale_buff",
+    "recurrence": { "math": [ "time_until('sunrise')" ] },
+    "global": true,
+    "run_for_npcs": true,
+    "condition": { "and": [ "u_is_outside", "u_can_see" ] },
+    "effect": [
+      {
```
Comment (Maleclypse) [inline data/json/effects_on_condition/misc_effect_on_condition.json:232]:
```suggestion
      { "u_message": "The sunrise always makes you a feel a little bit better.", "type": "morale" },
      {
```

File: data/json/effects_on_condition/misc_effect_on_condition.json [hunk 2]
```diff
@@ -220,5 +220,39 @@
...
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "sunset_morale_buff",
+    "recurrence": { "math": [ "time_until('sunset')" ] },
+    "global": true,
+    "run_for_npcs": true,
+    "condition": { "and": [ "u_is_outside", "u_can_see" ] },
+    "effect": [
+      {
```
Comment (Maleclypse) [inline data/json/effects_on_condition/misc_effect_on_condition.json:249]:
```suggestion
      { "u_message": "The sunset always makes you a feel a little bit better.", "type": "morale" },
      {
```

**Pull request merged**